### PR TITLE
feat(sql): add `lower()` and `upper()` string functions

### PR DESCRIPTION
**Pull request by blackstar-baba**
add lower and upper as aliases to our existing functions to_lower, to_upper to be PostgreSQL compatible which listed in issue #2578

### CODE REVIEW

**Comment by blackstar-baba** [comment]
Hi @pswu11, PTAL

File: core/src/main/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.str;
+
+public class LowerFunctionFactory extends ToLowercaseFunctionFactory{
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactory.java:27]: nit: weird formatting
```suggestion
public class LowerFunctionFactory extends ToLowercaseFunctionFactory {
```

File: core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Test
+    public void testWithNonAsciiCharacters() throws SqlException {
+        call("abcDEFghiJKLm...() { _; } >_[$($())] { <<< (=) \noPQRstuVXZ")
+                .andAssert("ABCDEFGHIJKLM...() { _; } >_[$($())] { <<< (=) \nOPQRSTUVXZ");
+    }
+
+    @Override
+    protected FunctionFactory getFunctionFactory() {
+        return new ToUppercaseFunctionFactory();
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java:61]:
```suggestion
        return new UpperFunctionFactory();
```

File: core/src/test/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    public void testWithNonAsciiCharacters() throws SqlException {
+        call("abcDEFghiJKLm...() { _; } >_[$($())] { <<< (=) \noPQRstuVXZ")
+                .andAssert("abcdefghijklm...() { _; } >_[$($())] { <<< (=) \nopqrstuvxz");
+    }
+
+    @Override
+    protected FunctionFactory getFunctionFactory() {
+        return new ToLowercaseFunctionFactory();
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactoryTest.java:62]:
```suggestion
        return new LowerFunctionFactory();
```

File: core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java (hunk 2 above)
Comment (blackstar-baba) [reply core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java:61]: my fault

File: core/src/test/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactoryTest.java (hunk 3 above)
Comment (blackstar-baba) [reply core/src/test/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactoryTest.java:62]: My fault, sorry

File: core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java (hunk 2 above)
Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/engine/functions/str/UpperFunctionFactoryTest.java:61]: No problem at all.

File: core/src/main/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (blackstar-baba) [reply core/src/main/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactory.java:27]: Do we have a template for comment?

[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/griffin/engine/functions/str/LowerFunctionFactory.java:27]: No strict template as such. I'd use it something like the following:
```
Postgres-compatibility `lower()` alias for the `to_lowercase()` function.
```

**Review by puzpuzpuz — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**